src/StudyRegistration/StudyRegistration.tsx

```diff
@@ -0,0 +1,376 @@
+import React, { useState, useEffect } from 'react';
+import {
+  Form,
+  Input,
+  Button,
+  Select,
+  Divider,
+  Tooltip,
+  Typography,
+  Space,
+  Result,
+} from 'antd';
+import { ResultStatusType } from 'antd/lib/result';
+import {
+  MinusCircleOutlined,
+  PlusOutlined,
+  QuestionCircleOutlined,
+} from '@ant-design/icons';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { Link, useLocation } from 'react-router-dom';
+
+import './StudyRegistration.css';
+import { userHasMethodForServiceOnResource } from '../authMappingUtils';
+import { useArboristUI, studyRegistrationConfig } from '../localconf';
+import loadStudiesFromMDS from '../Discovery/MDSUtils';
+import { registerStudyInMDS, preprocessStudyRegistrationMetadata, createCEDARInstance } from './utils';
+
+const { Option } = Select;
+
+export interface FormSubmissionState {
+  status?: ResultStatusType;
+  text?: string
+}
+export interface User {
+  username: string
+}
+export interface StudyRegistrationProps {
+  user: User,
+  userAuthMapping: any
+}
+interface LocationState {
+  studyUID?: string | number;
+}
+
+const layout = {
+  labelCol: {
+    span: 8,
+  },
+  wrapperCol: {
+    span: 32,
+  },
+};
+const tailLayout = {
+  wrapperCol: {
+    offset: 8,
+    span: 16,
+  },
+};
+const formItemLayoutWithOutLabel = {
+  wrapperCol: {
+    span: 16,
+    offset: 8,
+  },
+};
+/* eslint-disable no-template-curly-in-string */
+const validateMessages = {
+  required: '${label} is required',
+};
+/* eslint-enable no-template-curly-in-string */
+
+const handleClinicalTrialIDValidation = async (_, ctID: string): Promise<boolean|void> => {
+  if (!ctID) {
+    return Promise.resolve(true);
+  }
+  const resp = await fetch(`https://clinicaltrials.gov/api/query/field_values?expr=${encodeURIComponent(`SEARCH[Study](AREA[NCTId] ${ctID})`)}&field=NCTId&fmt=json`);
+  if (!resp || resp.status !== 200) {
+    return Promise.reject('Unable to verify ClinicalTrial.gov ID');
+  }
+  try {
+    const respJson = await resp.json();
+    if (respJson.FieldValuesResponse?.FieldValues?.length === 1 && respJson.FieldValuesResponse.FieldValues[0].FieldValue === ctID) {
+      return Promise.resolve(true);
+    }
+    return Promise.reject('Invalid ClinicalTrial.gov ID');
+  } catch {
+    return Promise.reject('Unable to verify ClinicalTrial.gov ID');
+  }
+};
+
+const isUUID = (input: string) => {
+  // regexp for checking if a string is possibly an UUID, from https://melvingeorge.me/blog/check-if-string-valid-uuid-regex-javascript
+  const regexp = /^[0-9a-fA-F]{8}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{12}$/gi;
+  return new RegExp(regexp).test(input);
+};
+
+const handleUUIDValidation = (_, UUID: string): Promise<boolean|void> => {
+  if (!UUID || !isUUID(UUID)) {
+    return Promise.reject('Invalid CEDAR user UUID');
+  }
+  return Promise.resolve(true);
+};
```

Comment:
```suggestion
const handleUUIDValidation = (_, UUID: string): Promise<boolean|void> => {
  if (UUID && isUUID(UUID)) {
    return Promise.resolve(true);
  }
  return Promise.reject('Invalid CEDAR user UUID');
};
```
current code looks to validate anything as good, may need to move things around for it to work, see ^^